nova_vm/src/ecmascript/builtins/keyed_collections/set_objects/set_constructor.rs

```diff
@@ -107,11 +107,11 @@ impl SetConstructor {
             }
         }
         // 7. Let iteratorRecord be ? GetIterator(iterable, SYNC).
-        let iterator_record = get_iterator(agent, iterable, false)?;
+        let mut iterator_record = get_iterator(agent, iterable, false)?;
```

Comment:
**thought:** We should probably split sync and async iterator creator functions from one another. There's some code duplication but the late bool check inside the functions hardly makes sense.